hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/table/functional/TestHoodieSparkMergeOnReadTableCompaction.java

```diff
@@ -163,6 +173,85 @@ public void testWriteDuringCompaction(String payloadClass) throws IOException {
     assertEquals(300, readTableTotalRecordsNum());
   }
 
+  @Test
+  public void testOutOfOrderCompactionSchedules() throws IOException, ExecutionException, InterruptedException {
```

Comment:
Similar to this, do we also need a test for out of order clean and cluster schedules?

Reply:
I have added this in PR desc 

```
This patch is introducing an additional step to validate the the instant time used to generate the plan is the highest and there is no other instant time in the timeline whose time > current instant time of interest.
Ideally, we might need to take table lock before doing this timestamp validation. But it expands the scope of the patch to take locks for every operation. With this patch, we are locking while generating instants for ingestion writes, compaction and clustering. For cleaning and rollbacks, we plan to take it up later.
```

So, clean and rollbacks are not taking locks here. and atleast for clean, strictly speaking it does not matter. no other table service depends on cleans. only subsequent cleans depends on previous clean.

Reply:
https://issues.apache.org/jira/browse/HUDI-8240